### PR TITLE
Extended Agent telemetry histogram details

### PR DESCRIPTION
**Pull request by iglendd**
### What does this PR do?
Extended Agent telemetry histogram details, specifically
- Added to a histogram's payload previously omitted and implicit `+Inf` bucket value
- Added histogram's p75, p95 and p99 values (expressed as the upper-bound for the matching bucket).

### Motivation
Make it more accurate and easier to use

### Describe how you validated your changes

Unit tests have been added [here](https://github.com/DataDog/datadog-agent/pull/32343/files#diff-61c8fdfafcc996c03c7ebc1633a81b62536c6d72b5b5c90abccc73bc2b95e30e)

### CODE REVIEW

File: releasenotes/notes/agent-tel-extend-histogr-6e2da94e63edcaf8.yaml [hunk 1]
```diff
@@ -0,0 +1,14 @@
...
+# it does not depend on any information only available in another
+# section. This may mean repeating some details, but each section
+# must be readable independently of the other.
+#
+# Each section note must be formatted as reStructuredText.
+---
+enhancements:
+  - |
+    Extended Agent telemetry histogram details, specifically:
+      - Added to a histogram's payload previously omitted and implicit `+Inf` bucket value
```
Comment (hestonhoffman) [inline releasenotes/notes/agent-tel-extend-histogr-6e2da94e63edcaf8.yaml:12]:
```suggestion
      - Added previously omitted and implicit `+Inf` bucket value to histogram payload 
```
Not sure if this changes the meaning. I found the original sentence a little hard to parse

File: releasenotes/notes/agent-tel-extend-histogr-6e2da94e63edcaf8.yaml [hunk 2]
```diff
@@ -0,0 +1,14 @@
...
+      - Added histogram's p75, p95 and p99 values (expressed as the upper-bound for the
```
Comment (hestonhoffman) [inline releasenotes/notes/agent-tel-extend-histogr-6e2da94e63edcaf8.yaml:13]:
```suggestion
      - Added p75, p95, and p99 histogram values (expressed as the upper-bound for the
```

**Review by hestonhoffman — COMMENTED** [review]
Couple suggestions

File: comp/core/agenttelemetry/impl/agenttelemetry.go [hunk 3]
```diff
@@ -276,44 +276,52 @@ func buildKeysForMetricsPreviousValues(mt dto.MetricType, metricName string, met
...
 			// add bucket names to the key
 			for _, bucket := range m.Histogram.GetBucket() {
 				keyNames = append(keyNames, fmt.Sprintf("%v:%v", keyName, bucket.GetUpperBound()))
 			}
-		} else {
-			keyNames = append(keyNames, keyName)
 		}
+
+		// For regular metric (and for HISTOGRAM +Inf bucket which follows the last bucket)
+		keyNames = append(keyNames, keyName)
```
Comment (dustmop) [inline comp/core/agenttelemetry/impl/agenttelemetry.go:288]: Should this be inside the for loop that starts at line 260? If `metrics` has 2 items (for example) and neither has tags, then `keyName` gets set to the `metricName` both iterations of the loop, and then appended to `keyNames` twice.

Comment (iglendd) [reply comp/core/agenttelemetry/impl/agenttelemetry.go:284]: Indeed @dustmop that would be a problem. And visually it looks like that, but if you look carefully the line is inside the loop. Still, thank you for noticing that, I fall to the same conclusion by looking on the diff here.

At any rate, I have used this as an opportunity to improve code a bit more and write a few more unit tests. Thank you again.

**Review by dustmop — APPROVED** [review]

**Comment by iglendd** [comment]
Thank you @dustmop. Sorry for adding to the confusion. I have added more comments which possibly clarify it. Indeed there are 2 possibilities (with tags and without) doubled by histogram. See my last commit for the comments.

File: releasenotes/notes/agent-tel-extend-histogr-6e2da94e63edcaf8.yaml [hunk 4]
```diff
@@ -0,0 +1,14 @@
...
+# it does not depend on any information only available in another
+# section. This may mean repeating some details, but each section
+# must be readable independently of the other.
+#
+# Each section note must be formatted as reStructuredText.
+---
+enhancements:
+  - |
+    Extended Agent telemetry histogram details, specifically:
+      - Added previously omitted and implicit `+Inf` bucket value to histogram payload
```
Comment (estherk15) [inline releasenotes/notes/agent-tel-extend-histogr-6e2da94e63edcaf8.yaml:12]:
```suggestion
      - Added previously omitted and implicit `+Inf` bucket value to histogram payload.
```

**Review by estherk15 — APPROVED** [review]
Non blocking suggestion
